use-shopping-cart/src/index.test.js

```diff
@@ -558,3 +558,115 @@ describe('stripe handling', () => {
     )
   })
 })
+
+function mockCartDetails(overrides) {
+  return {
+    [`sku_abc${counter}`]: {
+      sku: `sku_abc${counter++}`,
+      name: 'Bananas',
+      image: 'https://blah.com/banana.avif',
+      price: 400,
+      currency: 'USD',
+      value: 800,
+      quantity: 2,
+      formattedValue: '$8.00',
+      ...overrides
+    },
+    [`sku_efg${counter}`]: {
+      sku: `sku_efg${counter++}`,
+      name: 'Oranges',
+      image: 'https://blah.com/orange.avif',
+      currency: 'USD',
+      price: 250,
+      value: 1000,
+      quantity: 4,
+      formattedValue: '$10.00',
+      ...overrides
+    }
+  }
+}
+
+describe('loadCart()', () => {
+  it('should add cartDetails to cart object', async () => {
+    const wrapper = createWrapper()
+    const cart = renderHook(() => useShoppingCart(), { wrapper }).result
+    const cartDetails = mockCartDetails()
+    const product = mockProduct({ price: 200 })
+
+    act(() => {
+      cart.current.addItem(product)
+      cart.current.loadCart(cartDetails, false)
+    })
+
+    expect(cart.current.cartDetails).toEqual(cartDetails)
+    expect(cart.current.totalPrice).toEqual(1800)
+    expect(cart.current.cartCount).toEqual(6)
+  })
+
+  it('should merge two cart details items by default', async () => {
+    const wrapper = createWrapper()
+    const cart = renderHook(() => useShoppingCart(), { wrapper }).result
+
+    const cartDetails = mockCartDetails()
+    const product = mockProduct({ price: 200 })
+
+    act(() => {
+      cart.current.addItem(product)
+      cart.current.incrementItem(product.sku)
+      cart.current.loadCart(cartDetails)
+    })
+
+    const entry = cart.current.cartDetails[product.sku]
+    expect(cart.current.cartDetails).toEqual({
+      [entry.sku]: entry,
+      ...cartDetails
+    })
+    expect(cart.current.totalPrice).toEqual(2200)
+    expect(cart.current.cartCount).toEqual(8)
+  })
+})
+
+describe('filterCart', () => {
+  const wrapper = createWrapper()
+  const cart = renderHook(() => useShoppingCart(), { wrapper }).result
+  it('filter out all cart items that fail filter callback', async () => {
+    const cartDetails = {
+      'test-sku-123': {
+        sku: 'test-sku-123',
+        name: 'Bananas',
+        image: 'https://blah.com/banana.avif',
+        price: 200,
+        currency: 'USD',
+        value: 400,
+        quantity: 2,
+        formattedValue: '$4.00'
+      },
+      'test-sku-234': {
+        sku: 'test-sku-234',
+        name: 'Oranges',
+        image: 'https://blah.com/orange.avif',
+        currency: 'USD',
+        price: 250,
+        value: 1000,
+        quantity: 4,
+        formattedValue: '$10.00'
+      }
+    }
+
+    const mockFilter = (item) => item.value < 500
+    const filteredCart = await cart.current.filterCart(cartDetails, mockFilter)
```

Comment:
The way the code is currently, it looks like `filterCart()` is not actually returned from `useShoppingCart()`. It seems to be exported from `src/utils.js` since it doesn't depend on any information from `useShoppingCart()`.

Reply:
@ChrisBrownie55 in the example you gave, I saw:

`import {filterCart} from 'use-shopping-cart' so I actually imported `filterCart` into the index.js and exported it as part of `use-shopping-cart`, so this does work.

Would you rather it be separate from the core `USC` api because you're right, it doesn't actually depend on information from the cart, or would you be okay with keeping it the same?

Reply:
Oh I see, yeah I missed that part (checking this on the GitHub Android app). Yeah I think it'd make more sense to keep it separate from the actual useShoppingCart function because including it gives the sense that it is a method with side effects.

Plus, it'd be much better if we could also import this function server-side and filter the cart before sending it to the client.

Example:
Client (initial request for site) -> Server
- Server gets cart from DB
- Server does `filterCart()`
- Server passes filtered cart to App in SSR setup
Client <- (filtered cart + web page via ssr) Server

Reply:
@ChrisBrownie55 sounds good to me! I'm gonna move it around now